client/consensus/manual-seal/src/consensus/babe.rs

```diff
@@ -122,14 +126,45 @@ impl<B, C> ConsensusDataProvider<B> for BabeConsensusDataProvider<B, C>
 			})?;
 
 		// this is a dev node environment, we should always be able to claim a slot.
-		let (predigest, _) = authorship::claim_slot(slot_number, epoch.as_ref(), &self.keystore)
-			.ok_or_else(|| Error::StringError("failed to claim slot for authorship".into()))?;
-
-		Ok(Digest {
-			logs: vec![
+		let logs =  if let Some((predigest, _)) = authorship::claim_slot(slot_number, epoch.as_ref(), &self.keystore) {
+			vec![
 				<DigestItemFor<B> as CompatibleDigestItem>::babe_pre_digest(predigest),
-			],
-		})
+			]
+		} else {
+			drop(epoch);
+			// well we couldn't claim a slot because this is an existing chain and we're not in the authorities.
+			// we need to tell BabeBlockImport that the epoch has changed, and we put ourselves in the authorities.
+			let predigest = PreDigest::SecondaryPlain(SecondaryPlainPreDigest {
+				slot_number,
+				authority_index: 0_u32,
+			});
+
+			let authority = (AuthorityId::from(Alice.public()), 1000);
+
+			let epoch_mut = match epoch_descriptor {
+				ViableEpochDescriptor::Signaled(identifier, _epoch_header) => {
+					epoch_changes.epoch_mut(&identifier)
+						.ok_or_else(|| sp_consensus::Error::InvalidAuthoritiesSet)?
+				},
+				_ => unreachable!("we couldn't claim a slot, so this isn't the genesis epoch; qed")
+			};
+
+			// mutate the current epoch
+			epoch_mut.authorities = vec![authority.clone()];
```

Comment:
I was actually hoping that we can pretend to be regular babe authorities instead of using Alice's key. Does this somehow leak to runtime? I.e. imagine that in the testing environment we would probably want to test that if one of the validators misbehaves Babe starts ignoring other blocks produced by that validator.
Is it required that this here is always `Alice`? Could we make this configurable?

Reply:
> I was actually hoping that we can pretend to be regular babe authorities instead of using Alice's key.

hmm, actually now that I think about it, we could probably just use secondary slots and never have to mutate the epoch. Will have to test this. But for now, will do this: 

> Could we make this configurable?